### PR TITLE
factor: Split numeric.rs into multiple modules

### PR DESCRIPTION
**Pull request by nbraud**
- [x] Keep individual file sizes manageable (numeric.rs was nearing 500 lines)
- [x] Make it easier to keep track of the tests for numeric's functionality.
- [x] Limit build.rs' dependency to `numeric::{modular_inverse, traits}`.
      The other modules can now use compile-time generated tables etc.

### CODE REVIEW

File: src/uu/factor/src/numeric/mod.rs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+// * (c) 2020 nicoo            <nicoo@debian.org>
+// *
+// * For the full copyright and license information, please view the LICENSE file
+// * that was distributed with this source code.
+
+mod gcd;
+pub use gcd::gcd;
+
+mod traits;
+use traits::{DoubleInt, Int, OverflowingAdd};
```
Comment (sylvestre) [inline src/uu/factor/src/numeric/mod.rs:12]: could you please fix this warning ?

Comment (nbraud) [reply src/uu/factor/src/numeric/mod.rs]: Done

Comment (nbraud) [reply src/uu/factor/src/numeric/mod.rs:12]: Reverted the “fix”, as the warning is erroneous.  *sigh*

Comment (nbraud) [reply src/uu/factor/src/numeric/mod.rs]: Fixed for good now.

**Pull request merged**